### ISSUE TITLE
Add Saturation and Value/Brightness random options to particles in addition to Hue

### ISSUE THREAD

**Issue by Arnklit** [godotengine/godot-proposals#10152]
### Describe the project you are working on

A game in production

### Describe the problem or limitation you are having in your project

You can randomize the Hue in the particles shaders, but not the Value/Brightness and Saturation

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Change the Hue Variation into HSV/HSB Variation and have 3 min/max value pairs and curves.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

![image](https://github.com/godotengine/godot-proposals/assets/4955051/74070668-5fbf-47b5-937e-33453a198586)

### If this enhancement will not be used often, can it be worked around with a few lines of script?

It would be used often

### Is there a reason why this should be core and not an add-on in the asset library?

It would have to be built in to be practical as it would otherwise be a custom shader without the built in inspector.

**Comment by QbieShay** [comment]
Hey! You say this "would be used often", yet it's not something I have the impression people would actually use. Value can be changed with the color curve, so the only one left is saturation, which seems a bit strange to want.

**Comment by Calinou** [comment]
Remember that if you want to customize individual particle colors with more control, you can always use the initial color curve mapped to have different saturations.

**Comment by Arnklit** [comment]
Yeah I think it works well with brightness using the Initial Ramp, but not for hue, because the Ramp is only multiplied with the particle texture, a low hue in the ramp won't de-saturate the particle texture.

But to be fair the random brightness was mostly what we needed, saturation is a bit more of an edge case.

**Comment by Calinou** [comment]
Changing the texture's saturation requires using a custom shader on the rendering material either way, so a ParticleProcessMaterial is unable to do this.

**Comment by Arnklit** [comment]
Closing this as I think the arguments that this is not needed are pretty clear.
